plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java

```diff
@@ -459,6 +461,447 @@ public void testArrayFields()
+                "    }" +
+                "  }," +
+                "  \"properties\": {" +
+                "    \"es_binary\": {" +
+                "      \"type\": \"binary\"" +
+                "    }," +
+                "    \"es_boolean\": {" +
+                "      \"type\": \"boolean\"" +
+                "    }," +
+                "    \"es_long\": {" +
+                "      \"type\": \"long\"" +
+                "    }," +
+                "    \"es_integer\": {" +
+                "      \"type\": \"integer\"" +
+                "    }," +
+                "    \"es_short\": {" +
+                "      \"type\": \"short\"" +
+                "    }," +
+                "    \"es_byte\": {" +
+                "      \"type\": \"byte\"" +
+                "    }," +
+                "    \"es_double\": {" +
+                "      \"type\": \"double\"" +
+                "    }," +
+                "    \"es_float\": {" +
+                "      \"type\": \"float\"" +
+                "    }," +
+                "    \"order_field\": {" +
+                "      \"type\": \"integer\"" +
+                "    }" +
+                "  }" +
+                "}";
+
+        createIndex(indexName, mapping);
+
+        index(indexName, ImmutableMap.<String, Object>builder()
+                .put("es_binary", "test".getBytes(UTF_8))
+                .put("es_boolean", true)
+                .put("es_long", (long) 123)
+                .put("es_integer", 123)
+                .put("es_short", (short) 123)
+                .put("es_byte", (byte) 123)
+                .put("es_double", (double) 123)
+                .put("es_float", (float) 123)
+                .put("order_field", 1)
+                .build());
+
+        MaterializedResult rows = computeActual("" +
+                "SELECT " +
+                "es_binary, " +
+                "es_boolean, " +
+                "es_long, " +
+                "es_integer, " +
+                "es_short, " +
+                "es_byte, " +
+                "es_double, " +
+                "es_float " +
+                "FROM " + indexName + " " +
+                "ORDER BY order_field");
+
+        MaterializedResult expected = resultBuilder(getSession(), rows.getTypes())
+                .row("\"dGVzdA==\"", "true", "123", "123", "123", "123", "123.0", "123.0")
+                .build();
+
+        assertEquals(rows.getMaterializedRows(), expected.getMaterializedRows());
+
+        deleteIndex(indexName);
+    }
+
+    @Test
+    public void testAsRawJsonCases()
+            throws IOException
+    {
+        String indexName = "raw_json_cases_" + randomTableSuffix();
+
+        @Language("JSON")
+        String mapping = "" +
+                "{" +
+                "  \"_meta\": {" +
+                "    \"presto\": {" +
+                "      \"es_binary\": {" +
+                "        \"asRawJson\": true" +
+                "      }," +
+                "      \"esBoolean\": {" +
```

Comment:
I meant testing a nested object with a field that is named with non-lowercase, so that we can identify if there're any bugs with decodercontext. we should be able to do that in the previous test itself.